lahja/common.py

```diff
@@ -0,0 +1,111 @@
+from abc import (
+    ABC,
+    abstractmethod,
+)
+from typing import (
+    TYPE_CHECKING,
+    Generic,
+    NamedTuple,
+    Optional,
+    Set,
+    Type,
+    TypeVar,
+    Union,
+)
+
+from lahja.misc import (
+    BroadcastConfig,
+)
+
+if TYPE_CHECKING:
+    from lahja.base import EndpointAPI
+
+
+class BaseEvent:
+    __slots__ = ("_origin", "_id")
+
+    @property
+    def is_bound(self) -> bool:
+        return hasattr(self, '_origin')
+
+    def bind(self,
+             endpoint: 'EndpointAPI',
+             id: Optional[str] = None) -> None:
+        """
+        Bind an event to an end
+        """
+        self._origin = endpoint.name
+        self._id = id
+
+    def response_config(self, internal: bool = False) -> BroadcastConfig:
```

Comment:
Wanting to rename this to `get_response_config`

Reply:
Fine by me in general but maybe `create_response_config` or `derive_response_config` would reflect better what it does? We aren't just accessing any internal thing here but instead create a `BroadcastConfig` based on this event to form a reply. No strong feelings though.